### ISSUE TITLE
[Bug]: Setting a category to enabled by default doesn't work in opt-in mode

### ISSUE THREAD

**Issue by MoritzLost** [orestbida/cookieconsent#538]
### Expected Behavior

I'm seeing an issue where setting a category to `enabled: true` does not take effect when `mode: 'opt-in'` is set.

```js
CookieConsent.run({
    mode: 'opt-in',
    categories: {
        analytics: {
            enabled: true,
        },
    },
}
```

Setting both `enabled: true` and `readOnly: true` sets the toggle in the preferences modal to active and prevents this from being changed:

![Screenshot 2023-06-22 at 11 48 06](https://github.com/orestbida/cookieconsent/assets/10146880/a94648af-8a1e-4706-84c9-a947982d4a8b)

However, setting _only_ `enabled: true` has no effect:

![Screenshot 2023-06-22 at 11 48 23](https://github.com/orestbida/cookieconsent/assets/10146880/8dbe13bc-65b5-4b71-9a3f-ff5c6d178078)

In this configuration, I would expect the toggle to be active by default, with the user being able to toggle it off.

### Current Behavior

Setting only `enabled: true` does not activate the toggle in the preferences by default.

### Steps to reproduce

Use the config above and open the preferences modal.

### Version

3.0.0-rc.15

### On which browser do you see the issue?

Firefox

**Comment by orestbida** [comment]
This needs to be documented!

When `mode: 'opt-in'`, categories may not be enabled without the user's explicit consent. The `enabled` option only works if `mode: 'opt-out'`.

**Comment by MoritzLost** [comment]
> When mode: 'opt-in', categories may not be enabled without the user's explicit consent. The enabled option only works if mode: 'opt-out'.

@orestbida This feels very counter-intuitive to me. I think the expected behaviour would be as follows:

- In `opt-out` mode, everything that's enabled by default will be immediately active until the user opts out.
- In `opt-in` mode, the `enabled` setting controls the default values for when the preferences are opened for the first time, then the user can change those defaults.

It could also be argued that `enabled` should basically take precedence over `opt-in` mode and should always activate those categories immediately, though I would prefer the behaviour mentioned above.

The only thing that doesn't make sense (to me) is completely ignoring the `enabled` config in `opt-in` mode. Is there any reason _not_ to use it in the way described above?

**Comment by orestbida** [comment]
Pre-ticked boxes are against GDPR. Also, having non-essential categories set as enabled by default (although they aren't really enabled) is misleading for the user. The plugin shall only reflect only a valid and clear state.

If you still insist on achieving this behavior, then you will need to use custom code, sorry.

**Comment by kpcftsz** [comment]
It might be against GDPR, but as far as I'm aware CCPA still allows for this kind of "opt the user in by default" behavior. There are plenty of websites that don't care about GDPR but still need to be CCPA compliant.

**Comment by orestbida** [comment]
@kpcftsz , in this case you can explicitly set `mode: 'opt-out'` and then enable any category. The user will have the option to opt-out in any moment.

@MoritzLost , the `enabled` and `mode` options were meant to work this way. Perhaps ditching `mode` and only relying on `enabled` would have been a better choice; The current behaviour cannot be changed though, as it would be quite a breaking change. Perhaps in a future v4.

**Comment by MoritzLost** [comment]
@orestbida Fair enough. Removing `mode` and being able to pre-select specific categories (and services) would be an improvement IMO.
